server/master.go

```diff
@@ -0,0 +1,187 @@
+package server
+
+import (
+	// "context"
+	"encoding/json"
+	"fmt"
+	"github.com/charmbracelet/log"
+	"net"
+	"os"
+	"sync"
+
+	// pb "github.com/bsach64/goback/server/backuptask"
+	"golang.org/x/crypto/ssh"
+	// "google.golang.org/grpc"
+	// "google.golang.org/grpc/credentials/insecure"
+)
+
+// Master server is defined using
+// 1. List of workers
+// 2. Index of last worker that was assigned task
+// 3. Host host for the Master currently (0.0.0.0)
+// 4. Port of the Master
+// 5. IdRsa since this is a SSH server the master should also have a IDRsa
+type Server struct {
+	workers []Worker
+	index   int
+	mu      sync.Mutex
+	Host    string
+	Port    int
+	IdRsa   string
+}
+
+// Creates a new master server at 0.0.0.0 at port 2022
+func NewMaster() {
+
+	//Master server
+	m := Server{
+		index: 0,
+		Host:  "0.0.0.0",
+		Port:  2022,
+		IdRsa: "private/id_rsa",
+	}
+
+	//Create a new worker to be changed later
+	err := StartNewWorker(&m, 1, 1, "127.0.0.1", 2025)
```

Comment:
Ideally worker creation should be client side?

Reply:
> Ideally worker creation should be client side?

Yes, the worker should be created alongside the client.

Ideally, the server should have 0 workers at first, and when a client connects to the server, server should append the worker's detail that is running in the client's device.